docs/rules/forbid-component-props.md

```diff
@@ -35,10 +35,18 @@ The following patterns are not considered warnings:
 
 ```js
 ...
-"forbid-component-props": [<enabled>, { "forbid": [<string>] }]
+"forbid-component-props": [<enabled>, { "forbid": [<string>], "forbidPatterns": [<string>], ignoreDomNodes: <boolean> }]
 ...
 ```
 
 ### `forbid`
 
 An array of strings, with the names of props that are forbidden. The default value of this option is `['className', 'style']`.
+
+### `forbidPatterns`
+
+An array of strings, with the patterns of props that are forbidden; you can use `*` as a wildcard. e.g: `['data-*', '*-foo-*']`
+
+### `ignoreDomNodes`
+
+Whether to check forbidden props on DOM nodes (div, input, span, etc...). Defaults to true
```

Comment:
having options where absence and `false` are not the same is a very bad idea - it'd be better to rename this to `checkDOMNodes` and default it to `false`.